data-collection/src/main/java/refactoringml/ProcessMetricsCollector.java

```diff
@@ -68,24 +68,47 @@ private void collectProcessMetricsOfRefactoredCommit(CommitMetaData superCommitM
 		}
 	}
 
+	//update the process metrics for all renames that were missed by RefactoringMiner
+	private void processRenames(Set<ImmutablePair<String, String>> refactoringRenames, Set<ImmutablePair<String, String>> jGitRenames, CommitMetaData superCommitMetadata) {
+		//get all renames not detected by RefactoringMiner
+		if(refactoringRenames != null){
+			for(ImmutablePair<String, String> rename : refactoringRenames){
+				//check if the class file name was changed, not only the class name
+				if (!rename.left.equals(rename.right)){
+					pmDatabase.renameFile(rename.left, rename.right, superCommitMetadata);
+					log.debug("Renamed " + rename.left + " to " + rename.right + " in PMDatabase.");
+				}
+			}
+		}
+
+		//process the renames missed by refactoringminer
+		if(jGitRenames != null){
+			//get all renames missed by RefactoringMiner
+			if(refactoringRenames != null)
+				jGitRenames.removeAll(refactoringRenames);
+			if(jGitRenames.size() > 0){
+				log.debug("Refactoringminer missed these refactorings: " + jGitRenames + LogUtils.createErrorState(superCommitMetadata.getCommitId(), project));
+				//update the missed renames in the PM database
+				for(ImmutablePair<String, String> rename : jGitRenames){
+					log.debug("Renamed " + rename.left + " to " + rename.right + " in PMDatabase.");
+					pmDatabase.renameFile(rename.left, rename.right, superCommitMetadata);
+					pmDatabase.reportRefactoring(rename.right, superCommitMetadata);
```

Comment:
are you sure you want to count this rename as a refactoring? I suggest we only trust on refactorings that come from RefactoringMiner, as it uses a bit more semantic, rather than simply file renames.

Reply:
I would still count this as a refactoring, because it is exactly what the rename refactoring describes. Also, they are pretty rare.